### PR TITLE
Update gce Containerd initialization to handle the runtime existence case

### PR DESCRIPTION
**Pull request by ibabou**
#### What type of PR is this?

/kind cleanup
/kind feature

#### What this PR does / why we need it:

Allowing startup script for windows to detect if Containerd runtime exists (like in a prebuilt image), and reconfigure only in this case.

```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 1]
```diff
@@ -1569,11 +1569,24 @@ function Install_Containerd {
   Remove-Item -Force -Recurse $tmp_dir
 }
 
+# Lookup the path of containerd config if exists, else returns a default.
+function Get_Containerd_ConfigPath {
+  $service = Get-WMIObject -Class Win32_Service -Filter  "Name='containerd'"
+  if (!($service -eq $null) -and
+      $service.PathName -match ".*\s--config\s*(\S+).*" -and
```
Comment (jeremyje) [inline cluster/gce/windows/k8s-node-setup.psm1:1576]: Would it be better to just assume the path is hard codes or pass the config through kubeenv?

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 2]
```diff
@@ -1584,13 +1597,16 @@ function Configure_Containerd {
...
+  # Do the registration only if the containerd service does not exist.
+  if (Get-WMIObject -Class Win32_Service -Filter  "Name='containerd'" -eq $null) {
+    Log-Output "Creating containerd service"
+    & containerd.exe --register-service --log-file ${env:LOGS_DIR}/containerd.log
+  }
+
   Log-Output "Starting containerd service"
-  Start-Service containerd
-}
+  Restart-Service containerd
```
Comment (jeremyje) [inline cluster/gce/windows/k8s-node-setup.psm1:1609]: Why was this changed to Restart-Service?

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 3]
```diff
@@ -1584,13 +1597,16 @@ function Configure_Containerd {
...
 
-# Register and start containerd service.
+# Register if needed and start containerd service.
 function Start_Containerd {
-  Log-Output "Creating containerd service"
-  & containerd.exe --register-service --log-file ${env:LOGS_DIR}/containerd.log
+  # Do the registration only if the containerd service does not exist.
+  if (Get-WMIObject -Class Win32_Service -Filter  "Name='containerd'" -eq $null) {
+    Log-Output "Creating containerd service"
+    & containerd.exe --register-service --log-file ${env:LOGS_DIR}/containerd.log
```
Comment (jeremyje) [inline cluster/gce/windows/k8s-node-setup.psm1:1605]: Should the path be quoted?

**Review by jeremyje — CHANGES_REQUESTED** [review]

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 2 above)
Comment (ibabou) [reply cluster/gce/windows/k8s-node-setup.psm1:1609]: Because it was assuming that the runtime is always installed fresh through the script, now the service could already be existing and running, so we'll need to have a restart after updating its config.

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 3 above)
Comment (ibabou) [reply cluster/gce/windows/k8s-node-setup.psm1:1605]: It have been already as is, and it'll work except if the LOGS_DIR has spaces or any special chars. Will definitely have it quoted, thanks for the catch!

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 1 above)
Comment (ibabou) [reply cluster/gce/windows/k8s-node-setup.psm1:1576]: I guess we can do so, but I feel that assuming the config location never changes isn't perfect especially with containerd. unlike docker, where its installed package always kept the binaries in well-known paths, containerd is shipped as tar and can be extracted anywhere. This makes it a bit safer, so we don't re-register service again but knowing that a certain config location is what being used. If non is found, then it'll be the default path hardcoded by containerd.

**Review by jeremyje — APPROVED** [review]
